pkgs/tools/misc/hdf5/default.nix

```diff
@@ -23,8 +23,11 @@
 }:
 
 # cpp and mpi options are mutually exclusive
-# (--enable-unsupported could be used to force the build)
+# "-DALLOW_UNSUPPORTED=ON" could be used to force the build.
 assert !cppSupport || !mpiSupport;
+# cpp and threadsafe are mutually exclusive
+# "-DALLOW_UNSUPPORTED=ON" could be used to force the build.
+assert !cppSupport || !threadsafe;
```

Comment:
```suggestion
```
We do not need that assert here. CMake will complain about it and fail if built with both turned on.
If we keep that assert `-DALLOW_UNSUPPORTED=ON` is not even an option, since we get an evaluation failure,
before we even get to `cmakeConfigPhase`.
(The above cpp/mpi-related assert is old code that we should remove at some point for the same reason).